### PR TITLE
Nighthom model pipeline refactoring

### PR DESCRIPTION
**Pull request by Nighthom**
## 📌 변경 개요

> yolo_model.py 단일 파일을 레이어드 모듈로 분리하고, 스크립트·테스트·문서 전반을 정비합니다.

---

## 📋 변경 유형

- [ ] 🐛 버그 수정
- [ ] ✨ 새 기능 / 모듈 추가
- [ ] 🔬 실험 (exp/ 브랜치에서 main 병합 시)
- [ ] 📊 데이터 전처리 / 증강
- [x] 📝 문서 / 주석
- [ ] 🔧 설정 / 환경
- [x] ♻️ 리팩토링

---

## 📝 변경 내역
  - `yolo_model.py` 단일 파일을 YOLOModel / Trainer / Predictor / utils 레이어로 분리
  - `validate.py` 스크립트 추가 (val set mAP 독립 측정용)
  - 테스트 파일 구조 재편 및 커버리지 보강 (test_trainer, test_predictor, test_config 신규)
  - scripts / src / tests README 전면 개정 및 config 참조 문서(docs/config_reference.md) 신규 작성

---

## 🧪 테스트 / 검증 방법

> 변경 사항을 어떻게 검증했는지 설명하세요.

```bash
python -m pytest -v  
```
테스트에 대한 구체적인 내용은 tests/README.md를 확인해 주세요.

- [x] 로컬에서 직접 실행 확인
- [ ] Colab / Kaggle Kernel에서 실행 확인 (해당 시)

---

## ⚠️ 리뷰어 주의사항

> 특별히 집중해서 봐야 할 부분을 기술하세요.

---

## ✅ PR 제출 전 체크리스트

- [x] `ruff check .` 통과
- [x] `black --check .` 통과
- [x] 새 모듈에 docstring 작성
- [x] 예외 처리 포함 여부 확인
- [x] `.gitignore` 대상 파일(가중치, 이미지, 로그)이 포함되지 않았는지 확인
- [x] `data/raw/` 원본 데이터 수정 없음 확인
- [x] 금지 데이터(`TL_2_조합.zip`, `TS_2_조합.zip`) 미사용 확인

---

## 💬 기타

>

### CODE REVIEW

File: experiments/exp_20260420_baseline_yolo26n/config.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ val:
   conf: 0.25
   iou: 0.45
   max_det: 300
+  tta: false
```
Comment (Yopkigom) [inline experiments/exp_20260420_baseline_yolo26n/config.yaml:38]: tta 는 여기에서 false 입니다

File: scripts/predict.py [hunk 2]
```diff
@@ -14,30 +14,52 @@
...
     parser.add_argument("--source", required=True, help="이미지 디렉터리 경로")
     parser.add_argument(
         "--output",
-        default="results/predictions.json",
-        help="predictions.json 저장 경로",
+        default=None,
+        help="predictions.json 저장 경로 (미지정 시 config 우선순위 적용)",
+    )
+    parser.add_argument(
+        "--tta", action="store_true", help="TTA 활성화 (config val.tta보다 우선)"
```
Comment (Yopkigom) [inline scripts/predict.py:38]: 여기서 설정되는 argument 는 tta 를 활성화 한다고 되어 있습니다

File: experiments/exp_20260420_baseline_yolo26n/config.yaml (hunk 1 above)
Comment (Nighthom) [reply experiments/exp_20260420_baseline_yolo26n/config.yaml:38]: 네 원래 기본적으로 false가 맞고 --tta 플래그를 줘서 true로 키거나 config로 true를 키는걸 의도했습니다.

File: tests/test_trainer.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    trainer = Trainer(_make_model(sample_config))
+    kwargs = trainer._build_train_kwargs("data/processed/dataset.yaml", resume=False)
+
+    assert kwargs["device"] == "cuda:0"
+    assert list(kwargs.keys()).count("device") == 1
+
+
+def test_build_train_kwargs_seed_from_config(sample_config):
+    """seed가 cfg에서 올바르게 주입된다."""
+    sample_config["seed"] = 99
```
Comment (Yopkigom) [inline tests/test_trainer.py]: seed 값은 전역 변수 설정이 필요해보입니다

**Review by Yopkigom — APPROVED** [review]
모델을 Predictor와 Trainer 로 나누고 겹치는 부분을 공용화 한 부분과 제출 스크립트 정비한 내용 확인했습니다.
